config/routes.rb

```diff
@@ -290,6 +290,9 @@
           get  :remove_form
         end
       end
+      namespace :site_notices do
+        get :new
+      end
```

Comment:
This seems strange. I'd think that this route would be added automatically by the `resources :site_notices ` line below.

Also the remove_notice and dismiss_notice routes below can probably be removed since they are now handled through the API.

Reply:
Thanks. I removed the two routes handled through the API now.

I don't understand why exactly, but lines 293-295 need to be there to handle the case of returning an error if the submitted notice has no content. 

In line 36 of the new app/controllers/api/v1/site_notices_controller.rb, it redirects you to `admin_site_notices_new_path` if you submit a blank notice. If I remove lines 293-295 from routes.rb, that causes an `undefined local variable or method 'admin_site_notices_new_path'` error when you submit a blank notice.

I'm probably ignorantly doing something wrong in that error handling and/or the routes setup. Or maybe there's a better way to handle the blank notice problem. I'll try to work that out, but if you have any suggestions, please let me know.

Reply:
I'd try running `rake routes`. This will print out the built route table along with the names for each route. You take this name and add the `_path` suffix to get the function you should be calling.  I think what you'll find is that you should be using `new_admin_site_notice_path`  instead of `admin_site_notice_new_path`

There is more about all of this here: https://guides.rubyonrails.org/v3.2.13/routing.html

Reply:
@scytacki Thanks. What do you think of the latest changes?